PC/msvcrtmodule.c

```diff
@@ -601,10 +621,12 @@ PyInit_msvcrt(void)
     insertint(d, "LK_NBRLCK", _LK_NBRLCK);
     insertint(d, "LK_RLCK", _LK_RLCK);
     insertint(d, "LK_UNLCK", _LK_UNLCK);
+#ifndef MS_WINDOWS_GAMES
```

Comment:
These flags are needed for `SetErrorMode()`. Weirdly "winbase.h" only defines them for desktop apps, but `SetErrorMode()` is available for the app, system, and games API partitions. It seems like a miscommunication between the SDK developers. Can you try manually defining them and check whether `msvcrt.SetErrorMode()` works?

```c
#ifndef SEM_FAILCRITICALERRORS
#define SEM_FAILCRITICALERRORS      0x0001
#endif
#ifndef SEM_NOGPFAULTERRORBOX
#define SEM_NOGPFAULTERRORBOX       0x0002
#endif
#ifndef SEM_NOALIGNMENTFAULTEXCEPT
#define SEM_NOALIGNMENTFAULTEXCEPT  0x0004
#endif
#ifndef SEM_NOOPENFILEERRORBOX
#define SEM_NOOPENFILEERRORBOX      0x8000
#endif
```

A simple test to raise a system access violation using ctypes: `ctypes.c_int.from_address(0).value`. 

By default, Windows Error Reporting (WER) should quietly log the crash, but the system should attach a configured post-mortem debugger (e.g. Visual Studio; or WinDbg via `windbg -I`). Setting `msvcrt.SetErrorMode(msvcrt.SEM_NOGPFAULTERRORBOX)` should skip WER and skip attaching a post-mortem debugger. Python should simply crash with the exit status `STATUS_ACCESS_VIOLATION` (0xC0000005, or -1073741819).

Reply:
I am honestly unsure how to add `ctypes` to our xbox build. Our build does the following:
- build python_core
- freeze all python files
- build our game engine with embedded Python

From a quick check I did not see how to include ctypes in there.
So I performed the test differently:
```python
import msvcrt
for i in {0x1, 0x2, 0x4, 0x8000}:
    msvcrt.SetErrorMode(i)
```
does not lead to any errors. When running
```cpp
SetErrorMode(2);
int* test = nullptr;
*test = 1;
```
my visual studio debugger still breaks after the segfault.

Reply:
> my visual studio debugger still breaks after the segfault.

If your debugger is already attached, then it will always break. The change is in what happens without a debugger attached.

> I am honestly unsure how to add ctypes to our xbox build

Then leave it out. `ctypes` is basically optional, and the only thing in the stdlib that will break without it is some of the test suite.

I've been going through other major libraries getting fixes made where they had (unnecessary) ctypes dependencies, so it shouldn't be a horrible time to go without it.

Reply:
> If your debugger is already attached, then it will always break. The change is in what happens without a debugger attached.

Then I guess it simply does not really matter on the xbox + from the top of my head I have no clue how I would test this. Still since the `SetErrorMode` call passes it's fine for me either way.

> I've been going through other major libraries getting fixes made where they had (unnecessary) ctypes dependencies, so it shouldn't be a horrible time to go without it.

Yes I was just talking about the test. Our Windows/Linux/Playstation builds already run fine without ctypes :)